src/hosted-buttons/utils.js

```diff
@@ -77,8 +78,48 @@ export const createAccessToken: CreateAccessToken = memoize<CreateAccessToken>(
   }
 );
 
-const getButtonVariable = (variables: ButtonVariables, key: string): string =>
-  variables?.find((variable) => variable.name === key)?.value ?? "";
+/**
+ * Takes the preferences from the /ncp/api/form-fields response and
+ * turns it into an ordered array of preferred and eligible buttons.
+ */
+export const getEligibleButtonPreferences = (
+  preferences: HostedButtonPreferences
+): $ReadOnlyArray<string> => {
+  const {
+    button_preferences: buttonPreferences,
+    eligible_funding_methods: eligibleFundingMethods,
+  } = preferences;
+
+  if (!buttonPreferences.length) {
+    return eligibleFundingMethods;
+  }
+
+  const mutableEligibleFundingMethods = [...eligibleFundingMethods];
+  const preferredEligibleButtons = [];
+
+  for (let button of buttonPreferences) {
+    let buttonIndex = mutableEligibleFundingMethods.indexOf(button);
+    const isButtonEligible = buttonIndex >= 0;
+    const isDefault = button === "default";
+
+    // If button is ineligible or is "default", do nothing.
+    if (!isButtonEligible || isDefault) {
```

Comment:
## 🔮 A NOTE TO THE FUTURE 
Hey, hope you're doing well. 
If we ever want to allow more than two buttons, we _may_ have to change this logic. 
### Why
Think about the case where a merchant's button preferences were:
```
{
    button_preferences: ["paypal", "default", "venmo"],
    eligible_funding_methods: ["paypal", "venmo", "paylater"]
}
```

We would have to determine whether the output of `getEligibleButtonPreferences()` should be:

A. `["paypal", "paylater", "venmo"]` - Fill "default" with the next eligible funding method that is not in button_preferences, keeping the order of the specified buttons.

B. `["paypal", "venmo", "paylater"]` -  "default" uses the next eligible funding method, regardless of if its in button_preferences or not.

### For today
Since we only support two buttons right now, and the NCP team mentioned that they don't foresee adding support for more buttons in the near future, I don't want to add additional logic for a hypothetical future improvement.  We will cross that bridge when we get there, but it shouldn't be hard to add the logic here.  Its just a matter of what we would want that logic to be.